google_forms_to_jira/program.py

```diff
@@ -1,59 +1,69 @@
-"""This program polls a Google Form for new responses and creates a Jira issue for each new response.
+"""Create Jira issues based on Google Forms responses.
 
-Workflow:
-1. Trigger: HTTP GET request.
-2. Poll Forms: Poll the Google Form for new responses.
-3. Create Jira Issue: For each new response, create a Jira issue with the response data.
+Atlassian Jira API documentation:
+- https://docs.autokitteh.com/integrations/atlassian/jira/python
+
+Google Forms API documentation:
+- https://docs.autokitteh.com/integrations/google/forms/events
 """
 
-import json
 import os
-import time
 
-import autokitteh
 from autokitteh import google
 from autokitteh.atlassian import jira_client
 
 
-POLL_INTERVAL = os.getenv("POLL_INTERVAL")
+JIRA_PROJECT_KEY = os.getenv("JIRA_PROJECT_KEY")
+
+jira = jira_client("jira_conn")
+
+
+def on_form_response(event):
+    print("Form response submitted:", event.data)
+    response_id = event.data.response.response_id
+
+    forms = google.google_forms_client("forms_conn").forms()
+    questions = forms.get(formId=event.data.form_id).execute().get("items", [])
+
+    answers = _summarize_form_response(event.data.response.answers, questions)
+
+    # Check if a Jira issue already exists for this response (i.e. response edited).
+    query = f"project = {JIRA_PROJECT_KEY} AND description ~ {response_id}"
+    issues = jira.jql(query + " ORDER BY created DESC")
+    if issues.get("total", 0) == 0:
+        _create_jira_issue(answers, response_id)
+    else:
+        _update_jira_issue(issues["issues"][0], answers, response_id)
```

Comment:
```suggestion
def on_form_response(event):
    data = event.data
    print("Form response submitted:", data)
    response_id = data.response.response_id

    forms = google.google_forms_client("forms_conn").forms()
    questions = forms.get(formId=data.form_id).execute().get("items", [])

    answers = _summarize_form_response(data.response.answers, questions)

    # Check if a Jira issue already exists for this response (i.e. response edited).
    query = f"project = {JIRA_PROJECT_KEY} AND description ~ {response_id}"
    issues = jira.jql(query + " ORDER BY created DESC")
    if issues.get("total", 0) == 0:
        _create_jira_issue(answers, response_id)
    else:
        _update_jira_issue(issues["issues"][0], answers, response_id)
```

Reply:
That's a topic to discuss with Miki. I'd rather save users the need to set `data = event.data`.